src/main/java/com/github/dockerjava/jaxrs/BuildImageCmdExec.java

```diff
@@ -137,7 +137,7 @@ private WebTarget writeMap(WebTarget webTarget, String name, Map<String, String>
         if (value != null && !value.isEmpty()) {
             try {
                 return webTarget.queryParam(name,
-                        URLEncoder.encode(MAPPER.writeValueAsString(value), "UTF-8"));
+                        URLEncoder.encode(MAPPER.writeValueAsString(value), "UTF-8").replaceAll("\\+", "%20"));
```

Comment:
afair commons has EscapeUtils that may do this with few other chars.

Reply:
AFAIK `EscapeUtils`  escapes only HTML entities. Commons Lang provides `URLCodec` which basically behaves the same as `URLEncoder` (space is converted to `+`).